### DISCUSSION TITLE
Unable to get MeetingId and teamsContext properties in a teams tab that has been added to a meeting

### DISCUSSION THREAD

**Discussion by mohammadamer** [microsoft/hack-together#60] [Q&A]
Unable to get MeetingId and teamsContext properties in a teams tab that has been added to a meeting. What could be wrong ?
Caould you please help ?

### Description
MeetingId in teamsContext is always null and almost all TeamsContext properties.

### To Reproduce

1. Create a C# team tab app in visual studio
2. Change the code a bit to retrieve current the meeting Id where the tab has been added.
3. when debugging no errors but almost all properties are null
4. See screenshot

### Expected
Expected behavior is to be able to retrieve the teams context properties and the meeting Id

### Screenshots
![222967967-70addbe9-e8aa-4e71-bbf1-6083b6bdb19d](https://user-images.githubusercontent.com/19314043/222973605-2c6849f1-4d37-4c98-816f-2fdcd8ab67fa.png)

Code changes to get meeting Id

```cs
await base.OnAfterRenderAsync(firstRender);

    if(firstRender)
    {
        await MicrosoftTeams.InitializeAsync();
        isInTeams = await MicrosoftTeams.IsInTeams();

        if (isInTeams)
        { 
            var teamsMeeting = await MicrosoftTeams.GetTeamsContextAsync(); //teamsmeeting properties always null
            if (!String.IsNullOrEmpty(teamsMeeting.MeetingId))
                meetingId = teamsMeeting.MeetingId;
            else
                meetingId = "";
        }
        else
        {
            errorMessage = "Not running in Microsoft Teams.";
        }

        isLoading = false;
        StateHasChanged();            
    }
```

**Comment by aycabas** [comment]
Hi @mohammadamer, have you tried getting the Teams meeting id through Microsoft Graph `onlineMeeting`? This way you may access to all the details of the meeting. Here is the documentation for the details, let me know if you need further assistance on this: https://learn.microsoft.com/en-us/graph/api/onlinemeeting-get?view=graph-rest-1.0&tabs=http

**Reply by mmsharepoint** [reply]
The meetingID from the teamsContext is not the onlineMeetingId but an encrypted version if the chatID
Yannick Reekmans figured this out once upon a time:
https://blog.yannickreekmans.be/get-full-meeting-details-in-a-teams-meetings-app-without-using-bot-sdk/

**Reply by mohammadamer** [reply]
Thanks for the explaination @mmsharepoint 
It took from me two days until I figured it out. :)

**Comment by JerryYangKai** [comment]
Tracking here https://github.com/OfficeDev/TeamsFx/issues/7953

**Comment by mmsharepoint** [comment]
I could only give you answers dealing with JavaScript. Didn't try to refactor that to C#, yet. Even in my Visual Studio projects.
Would that help?

**Reply by mohammadamer** [reply]
With the help of the [article](https://blog.yannickreekmans.be/get-full-meeting-details-in-a-teams-meetings-app-without-using-bot-sdk/) that you mentioned and the explanation in [this issue](https://github.com/microsoftgraph/microsoft-graph-docs/issues/11696) 
and also @ explanation. This would help :)

Thanks a lot @aycabas @mmsharepoint @JerryYangKai 
You are all awesome :)

**Reply by mmsharepoint** [reply]
Here is a TeamsToolkit VisualStudio C# (backend) sample dealing with the teamsContext in TeamsJS SDK v2 in general:
https://github.com/mmsharepoint/tab-sso-graph-upload-pdf-csharp

And here one of my Teams meeting app samples dealing with the meetingID itself. That sample is nodeJS (yoteams) based but I hope you should be able to put the pieces together
https://github.com/mmsharepoint/tab-meeting-details/

**Reply by mohammadamer** [reply]
@mmsharepoint 
Thanks a lot Markus. Appreciated!
